pkg/cache/cluster.go

```diff
@@ -141,19 +154,56 @@ func NewClusterCache(config *rest.Config, opts ...UpdateSettingsFunc) *clusterCa
 	return cache
 }
 
+type Resources struct {
+	sync.Map
+}
+
+func (r *Resources) StoreResources(key kube.ResourceKey, value *Resource) {
+	r.Store(key, value)
+}
+
+func (r *Resources) LoadResources(key kube.ResourceKey) (*Resource, error) {
```

Comment:
Same as above - panicking should be ok if something unexpected is in this private field. And the error is not checked in a few places below. Why return it then? :)